### DISCUSSION TITLE
Manual conversion of .ckpt to diffusers

### DISCUSSION THREAD

**Discussion by Miraculix200** [d8ahazard/sd_dreambooth_extension#102] [Q&A]
I'm trying to get the extension to run on free Google Colab with only 12GB RAM (not VRAM). When I click the "Create" button when creating a model from a .ckpt, the webui always crashes.

Maybe it would work if I manually convert the .ckpt to diffusers, before starting the webui, as there would be a little more free RAM.

What would be the best way to do this from a .ipynb notebook?

One of the things I tried for the start is this:

```
%pip install git+https://github.com/huggingface/diffusers

%cd /tmp
!wget https://raw.githubusercontent.com/huggingface/diffusers/main/scripts/convert_original_stable_diffusion_to_diffusers.py

!python convert_original_stable_diffusion_to_diffusers.py
```

But it gives me a `ImportError: cannot import name 'DPMSolverMultistepScheduler' from 'diffusers'` error. I'm neither used to coding in Python nor do I have any idea how Dreambooth/SD works beneath the surface, so it would take me a very long time to figure out how to get it running.

Any suggestions what I could try?

**Comment by Miraculix200** [comment]
I guess I could also simply download the diffusers from Huggingface. But maybe I want to do a training with a custom model in the future, for which no diffusers are available

**Comment by d8ahazard** [comment]
FWIW, my "convert.py" is really just a mash-up of the two scripts from here:

https://github.com/ShivamShrirao/diffusers/tree/main/scripts

![image](https://user-images.githubusercontent.com/1633844/201168634-6b759028-bac0-43cd-b3ac-9c487252fbd1.png)

You should just need to create a folder in /SD_WEBUI_DIR/models/dreambooth/MODEL_NAME/working - where working looks like so:

![image](https://user-images.githubusercontent.com/1633844/201168828-4de20d43-7383-4c85-afa1-c652410a75bb.png)

The config file I'm feeding it comes from this URL:

https://raw.githubusercontent.com/CompVis/stable-diffusion/main/configs/stable-diffusion/v1-inference.yaml

**Comment by Miraculix200** [comment]
Alright, I managed to convert .ckpt to diffusers, and added v1-inference.yaml and dome db_.json file.

This Google Colab notebook doesn't do that (yet?), but it can do Dreambooth training using the Dreambooth extension for the webui. Didn't thorougly test it yet, but I successfully trained a model with 100 training steps. The notebook will clone diffusers from Huggingsface instead of converting a .ckpt file, 

The webui will also crash at the end of the training, when trying to convert diffusers to a .ckpt file. So the notebook has a cell to convert it after the crash. 

https://colab.research.google.com/github/Miraculix200/StableDiffusionUI_Colab/blob/main/StableDiffusionUI_Colab.ipynb

**Comment by Miraculix200** [comment]
Maybe the extension could have the option to clone diffusers from Huggingsface instead of converting .ckpt files to diffusers
